### PR TITLE
sql: use descriptor collection when looking up fully qualified name

### PR DESCRIPTION
**Pull request by ajwerner**
Before this commit, looking up the fully qualified name would read from the KV
store directly. There's no good reason to do this.

Release note: None

### CODE REVIEW

File: pkg/sql/resolver.go [hunk 1]
```diff
@@ -419,18 +418,20 @@ func getDescriptorsFromTargetListForPrivilegeChange(
...
 	schemaID := desc.GetParentSchemaID()
-	schemaName, err := resolver.ResolveSchemaNameByID(ctx, p.txn, p.ExecCfg().Codec, desc.GetParentID(), schemaID)
+	resolvedSchema, err := p.Descriptors().ResolveSchemaByID(ctx, p.txn, schemaID)
 	if err != nil {
 		return nil, err
 	}
 	tbName := tree.MakeTableNameWithSchema(
 		tree.Name(dbDesc.GetName()),
-		tree.Name(schemaName),
+		tree.Name(resolvedSchema.Name),
```
Comment (knz) [inline pkg/sql/resolver.go:434]: Why not .GetName() here?

**Review by knz — APPROVED** [review]

**Review by thoszhang — COMMENTED** [review]
> Reviewed 1 of 1 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/57963)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajwerner)

___
*[pkg/sql/resolver.go, line 421 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/57963#-MObwwPBDQ01AiGai4wY:-MObwwPBDQ01AiGai4wZ:b1lrfxf) ([raw file](https://github.com/cockroachdb/cockroach/blob/c0c2961e709967555adf41df4604be899883d2a1/pkg/sql/resolver.go#L421)):*
> ```Go
> 	ctx context.Context, desc catalog.TableDescriptor,
> ) (*tree.TableName, error) {
> 	dbDesc, err := p.Descriptors().GetDatabaseVersionByID(ctx, p.txn, desc.GetParentID(), tree.DatabaseLookupFlags{
> ```

This changes the behavior of the method, right? You're now always getting the name from the leased descriptor (or the uncommitted version, but only if the previous write actually went through the collection, and we presently don't cache the descriptor on reads). Are you sure you want to do this for all the callers?

**Review by ajwerner — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/57963)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @knz and @lucy-zhang)

___
*[pkg/sql/resolver.go, line 421 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/57963#-MObwwPBDQ01AiGai4wY:-MOcKBjb9U2OZFR0q_eI:b-ilcik3) ([raw file](https://github.com/cockroachdb/cockroach/blob/c0c2961e709967555adf41df4604be899883d2a1/pkg/sql/resolver.go#L421)):*
<details><summary><i>Previously, lucy-zhang (Lucy Zhang) wrote…</i></summary><blockquote>

This changes the behavior of the method, right? You're now always getting the name from the leased descriptor (or the uncommitted version, but only if the previous write actually went through the collection, and we presently don't cache the descriptor on reads). Are you sure you want to do this for all the callers?
</blockquote></details>

Only exceedingly narrowly but yes, there is a case to be made that it would in the face of a concurrent rename to the parent schema or database. The solution to that is in https://github.com/cockroachdb/cockroach/pull/56663. I'll get that PR fixed up and then clarify the semantics.
___
*[pkg/sql/resolver.go, line 434 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/57963#-MObvVLOlHwdqb8RxFF9-r1-434:-MOcKrAUD4QbsKMfl6CD:b-p10h3s) ([raw file](https://github.com/cockroachdb/cockroach/blob/c0c2961e709967555adf41df4604be899883d2a1/pkg/sql/resolver.go#L434)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

Why not .GetName() here?
</blockquote></details>

The `ResolvedSchema` is a rather irritating abstraction we added to deal with the fact that some schemas do not have descriptors. These are, in particular, temporary schemas and the public schema. This struct has a name field that is always populated and is not an interface like the database and tables descriptors.

**Review by knz — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/57963)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajwerner and @lucy-zhang)

___
*[pkg/sql/resolver.go, line 434 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/57963#-MObvVLOlHwdqb8RxFF9-r1-434:-MOcM5CMBcmH-SwuXnIt:bq42jxz) ([raw file](https://github.com/cockroachdb/cockroach/blob/c0c2961e709967555adf41df4604be899883d2a1/pkg/sql/resolver.go#L434)):*
<details><summary><i>Previously, ajwerner wrote…</i></summary><blockquote>

The `ResolvedSchema` is a rather irritating abstraction we added to deal with the fact that some schemas do not have descriptors. These are, in particular, temporary schemas and the public schema. This struct has a name field that is always populated and is not an interface like the database and tables descriptors. 
</blockquote></details>

cool thanks for explaining :+1:

**Comment by knz** [comment]
I think that this can be revived since the other PR was merged?

**Comment by ajwerner** [comment]
Sure, TFTR!

bors r=knz

**Pull request merged**